tests/test_jasmine_fast_epgp.py

```diff
@@ -0,0 +1,150 @@
+    Y = w @ xs.transpose(0, 2, 1)  #(n_samples x n x m)
+    if Poisson:
+        Y = np.random.poisson(2 * (Y - np.amin(Y)))
+    else:
+        Y = Y + np.random.normal(0, 0.2, size=Y.shape)
+    print('Y:', Y.shape, np.std(Y), np.quantile(Y, 0.99))
+
+    data = torch.tensor(Y, device=device, dtype=torch.get_default_dtype())
+    # specify manifold, kernel and rdist
+    manif = mgp.manifolds.Euclid(m, dfit)
+    #kernel = mgp.kernels.Linear(n, dfit)
+    kernel = mgp.kernels.Linear(n,
+                                dfit,
+                                ard=True,
+                                learn_scale=False,
+                                Y=Y,
+                                Poisson=Poisson)
+
+    #latent distribution is itself a GP
+    #     lat_dist = mgp.rdist.lat_GP(manif,
+    #                                 m,
+    #                                 n_samples,
+    #                                 torch.Tensor(ts).to(device),
+    #                                 Y=Y,
+    #                                 initialization='fa')
+
+    lat_dist = mgp.rdist.fast_EP_GP(manif,
+                                    m,
+                                    n_samples,
+                                    torch.Tensor(ts).to(device),
+                                    Y=Y,
+                                    initialization='fa')
+
+    ###construct prior
+    lprior = mgp.lpriors.GP_full(dfit, m, n_samples, manif,
+                                 torch.Tensor(ts).to(device))
+    lprior = mgp.lpriors.Null(manif)
+
+    # generate model
+    if Poisson:
+        likelihood = mgp.likelihoods.NegativeBinomial(n, Y=Y)
+        #likelihood = mgp.likelihoods.Poisson(n)
+    else:
+        likelihood = mgp.likelihoods.Gaussian(n, Y=Y, d=dfit)
+    z = manif.inducing_points(n, n_z)
+    mod = mgp.models.SvgpLvm(n, m, n_samples, z, kernel, likelihood, lat_dist,
+                             lprior).to(device)
+
+    print(mod.lat_dist.name)
+    ### test that training runs ###
+    n_mc = 16
+
+    def cb(mod, i, loss):
+        if i % 5 == 0:
+            print('')
+        return
+
+    mgp.optimisers.svgp.fit(data,
+                            mod,
+                            optimizer=optim.Adam,
+                            n_mc=n_mc,
+                            max_steps=5,
+                            burnin=50,
+                            lrate=10e-2,
+                            print_every=5,
+                            stop=cb,
+                            analytic_kl=True)
+
+    try:
+        print('lat ell, scale:',
+              mod.lat_dist.f.ell.detach().flatten(),
+              mod.lat_dist.f.scale.detach().flatten())
+        print('prior:', mod.lprior.ell.detach().flatten())
+    except:
+        print('lat ell, scale:',
+              mod.lat_dist.ell.detach().flatten(),
+              mod.lat_dist.scale.detach().mean(0).mean(-1))
+
+    print('kernel:', (mod.kernel.input_scale.detach())**(-1))
```

Comment:
If you comment out this line the test should pass -- we re-structured our LVMs a bit so mod.kernel no longer exists